unit_tests/src/test_rc_data_obj.cpp

```diff
@@ -540,14 +552,206 @@ TEST_CASE("open,read,write,close")
+            const auto fd = rcDataObjOpen(&comm, &open_inp);
+            REQUIRE(fd > 2);
+            CHECK(INTERMEDIATE_REPLICA == replica::replica_status(comm, target_object, 0));
+            CHECK(WRITE_LOCKED == replica::replica_status(comm, target_object, 1));
+
+            // Get the file descriptor info from the opened replica and extract the replica token.
+            // We are not dealing in hierarchies here so that piece is not required here.
+            char* out = nullptr;
+            const auto json_input = nlohmann::json{{"fd", fd}}.dump();
+            const auto l1_desc_info = rc_get_file_descriptor_info(&comm, json_input.data(), &out);
+            REQUIRE(l1_desc_info == 0);
+            const auto token = nlohmann::json::parse(out).at("replica_token").get<std::string>();
+            std::free(out);
+
+            // Open the same replica on a different connection using the replica token and replica
+            // resource hierarchy. Ensure that the open is successful. This would fail if the wrong
+            // replica is targeted because no agent can open a replica which is write-locked, as the
+            // sibling replica on the default resource should be in this case.
+            {
+                irods::experimental::client_connection conn2;
+                RcComm& comm2 = static_cast<RcComm&>(conn2);
+
+                dataObjInp_t open_inp2{};
+                std::snprintf(open_inp2.objPath, sizeof(open_inp2.objPath), "%s", path_str.data());
+                open_inp2.openFlags = O_CREAT | O_WRONLY;
+                addKeyVal(&open_inp2.condInput, RESC_HIER_STR_KW, test_resc.data());
+                addKeyVal(&open_inp2.condInput, REPLICA_TOKEN_KW, token.data());
+
+                const auto fd2 = rcDataObjOpen(&comm2, &open_inp2);
+                REQUIRE(fd2 > 2);
+                CHECK(INTERMEDIATE_REPLICA == replica::replica_status(comm2, target_object, 0));
+                CHECK(WRITE_LOCKED == replica::replica_status(comm2, target_object, 1));
+
+                openedDataObjInp_t close_inp{};
+                close_inp.l1descInx = fd2;
+                REQUIRE(rcDataObjClose(&comm2, &close_inp) >= 0);
+            }
+
+            openedDataObjInp_t close_inp{};
+            close_inp.l1descInx = fd;
+            REQUIRE(rcDataObjClose(&comm, &close_inp) >= 0);
+
+            {
+                // ensure all system metadata were updated properly
+                const auto [replica_info, replica_lm] = replica::make_replica_proxy(comm, target_object, 0);
+                CHECK(replica_info.mtime() == replica_info.ctime());
+                CHECK(0 == static_cast<unsigned long>(replica_info.size()));
+                CHECK(GOOD_REPLICA == replica_info.replica_status());
+            }
+
+            {
+                // ensure all system metadata were updated properly
+                const auto [replica_info, replica_lm] = replica::make_replica_proxy(comm, target_object, 1);
```

Comment:
i only see the replica number is different here... they should both be marked good?   please add another few words to these two 'ensure' blocks to differentiate/confirm what should be happening.      it is because there was no writing done here?   so the original statuses are all restored?  aka both back to `GOOD_REPLICA`?

Reply:
Okay, I will add some clarifying comments. 

Correct, the system metadata should be unchanged save for the replica statuses - which should be restored - because no writing was done here, as you said.

Reply:
Done